pkg/provision/civo.go

```diff
@@ -69,41 +70,28 @@ func (p *CivoProvisioner) Status(id string) (*ProvisionedHost, error) {
 	}, nil
 }
 
-func (p *CivoProvisioner) Delete(id string) error {
-
-	apiURL := fmt.Sprint("https://api.civo.com/v2/instances/", id)
-
-	req, err := http.NewRequest(http.MethodDelete, apiURL, nil)
-	if err != nil {
-		return err
+// Delete terminates the exit node
+func (p *CivoProvisioner) Delete(config HostDeleteConfig) error {
+	var id string
+	var err error
+	if len(config.IP) > 0 {
```

Comment:
I would do this the opposite way, if an ID is given, use that (fewer API calls and faster), then if not set, use the IP.

Reply:
The ID and IP will never be set at the same time as I check for that here 

```
	if isNotSet(hostID) && isNotSet(hostIP) {
		return fmt.Errorf("give a valid --id or --ip for your host")
	} else if isSet(hostID) && isSet(hostIP) {
		return fmt.Errorf("cannot use --id and --ip at the same time")
	}
```

I can remove this check and just prefer ID